flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/FilterRunner.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.runtime
+
+import org.apache.flink.api.common.functions.util.FunctionUtils
+import org.apache.flink.api.common.functions.{FilterFunction, RichFilterFunction}
+import org.apache.flink.configuration.Configuration
+import org.apache.flink.table.codegen.Compiler
+import org.slf4j.LoggerFactory
+
+class FilterRunner[IN] (
```

Comment:
Yes, it won't work otherwise ;-) The reason is the following: Flink serializes user functions (i.e., the `ProcTimeInnerJoin` `CoProcessFunction`) with Java serialization. We would compile the code-gen'd function at compile time and pass the resulting object into the user function, the deserialization would fail, because the code-gen'd class is not known at the remote worked. The code-gen'd class was only added to the classloader of the client JVM but not in the worker JVM. By shipping a String and compiling in the worker, we ensure that all information is available at the worker. 

We could of course also ship the generated class and load it to the classloader of the worker before deserializing the user function (and the included code-gen'd object), but Flink does not have a mechanism for this yet. We might add it later at some point because it would also be interesting for code-gen'd serializers and comparators.